src/gui/tray/activitylistmodel.cpp

```diff
@@ -689,6 +689,15 @@ void ActivityListModel::removeActivityFromActivityList(const Activity &activity)
     }
 }
 
+void ActivityListModel::checkAndRemoveSeenActivities(const OCC::ActivityList &receivedActivities)
+{
+    for (const auto &activity : _finalList) {
+        if (activity._objectType == QStringLiteral("chat") && !receivedActivities.contains(activity)) {
```

Comment:
nitpick: I would rename `receivedActivities` to `newActivities`, my tired brain thought this was a store of previously received activities and was trying to find where this was set 🤦‍♂️